packages/blocks/src/page-block/edgeless/connector-manager.ts

```diff
@@ -0,0 +1,919 @@
+import type { ConnectorElement, SurfaceManager } from '@blocksuite/phasor';
+import {
+  almostEqual,
+  AStarAlgorithm,
+  Bound,
+  clamp,
+  ConnectorMode,
+  getBoundFromPoints,
+  type IConnector,
+  type IVec,
+  lineIntersects,
+  linePolygonIntersects,
+  Overlay,
+  polygonNearestPoint,
+  sign,
+  Vec,
+} from '@blocksuite/phasor';
+import { assertEquals, assertExists } from '@blocksuite/store';
+
+import {
+  type Connectable,
+  type TopLevelBlockModel,
+} from '../../__internal__/utils/types.js';
+import type { EdgelessPageBlockComponent } from './edgeless-page-block.js';
+import { isTopLevelBlock } from './utils/query.js';
+import type { Selectable } from './utils/selection-manager.js';
+export class ConnectionOverlay extends Overlay {
+  surface!: SurfaceManager;
+  points: IVec[] = [];
+  highlightPoint: IVec | undefined = undefined;
+  rect: Bound | undefined = undefined;
+  override render(ctx: CanvasRenderingContext2D): void {
+    const zoom = this.surface.viewport.zoom;
+    this.points.forEach(p => {
+      ctx.beginPath();
+      ctx.arc(p[0], p[1], 3 / zoom, 0, Math.PI * 2);
+      ctx.fillStyle = 'white';
+      ctx.strokeStyle = 'blue';
+      ctx.fill();
+      ctx.stroke();
+    });
+    if (this.highlightPoint) {
+      ctx.beginPath();
+      ctx.arc(
+        this.highlightPoint[0],
+        this.highlightPoint[1],
+        3 / zoom,
+        0,
+        Math.PI * 2
+      );
+      ctx.fillStyle = 'blue';
+      ctx.strokeStyle = 'blue';
+      ctx.fill();
+      ctx.stroke();
+    }
+    if (this.rect) {
+      ctx.beginPath();
+      ctx.rect(this.rect.x, this.rect.y, this.rect.w, this.rect.h);
+      ctx.fillStyle = 'rgba(211, 211, 211, 0.3)';
+      ctx.fill();
+    }
+  }
+  clear() {
+    this.points = [];
+    this.highlightPoint = undefined;
+    this.rect = undefined;
+  }
+}
+
+export class EdgelessConnectorManager {
+  private _algorithm: AStarAlgorithm | undefined;
+  private _connectionOverlay = new ConnectionOverlay();
+  constructor(private _edgeless: EdgelessPageBlockComponent) {
+    this._edgeless.surface.viewport.addOverlay(this._connectionOverlay);
+    this._connectionOverlay.surface = this._edgeless.surface;
+  }
+
+  searchConnection(point: IVec, excludedIds: string[] = []) {
```

Comment:
Please add necessary blank lines or split sub methods to improve the readability of this method.